runtime/frame_descriptors.c

```diff
@@ -26,6 +26,34 @@
 #include "caml/shared_heap.h"
 #include <stddef.h>
 
+struct caml_frame_descrs {
+  int num_descr;
+  int mask;
+  frame_descr** descriptors;
+  caml_frametable_list *frametables;
+  caml_frametable_list *zombies;
+  caml_plat_mutex mutex;
```

Comment:
I would like to see a documentation comment for `mutex` in the comment below. The documentation should answer two questions:

- What is `mutex` used for (what are the parts of the mutable state whose access must be protected by the mutex?)?
- Why are we using a mutex and not just a STW section as already used by the code? (We use a mutex-protected section to mark frames as zombies, but we only remove them later in a STW section. Using a mutex instead of a STW section directly allows performing removal in more context, this is an important aspect of the design that should be pointed out explicitly.)
-